onnx/version_converter/convert.h

```diff
@@ -0,0 +1,162 @@
+
+class DefaultVersionConverter : public BaseVersionConverter {
+  private:
+    bool DEBUG = false;
+
+    bool searchOpDomainMap(const std::unordered_map<std::string, std::map<
+      int64_t, const OpSchema*>>& op_domain_map, int64_t curr_version) const {
+      const auto& version_it = op_domain_map.find("");
+      return version_it != op_domain_map.end() &&
+          version_it->second.find(curr_version) !=
+          version_it->second.end();
+    }
+
+  public:
+    DefaultVersionConverter() {
+      // Register adapters to the version converter
+      registerAdapter(std::unique_ptr<Adapter>(new NoPreviousVersionAdapter("Cos",
+        OpSetID(7), OpSetID(6))));
+    }
+
+    ModelProto convert_version(
+        const ModelProto& mp_in,
+        const OpSetID& initial_version,
+        const OpSetID& target_version) const override {
+      const char* initial_domain = initial_version.domain().c_str();
+      const char* target_domain = target_version.domain().c_str();
+
+      ONNX_ASSERTM((strcmp(initial_domain, "") == 0 || strcmp(initial_domain,
+              "ai.onnx") == 0) && (strcmp(target_domain, "") == 0 || strcmp(
+                target_domain, "ai.onnx") == 0),
+          "Warning: default onnx version converter can only convert "
+          " between default domain opset versions ('' or 'ai.onnx')\n"
+          "Provided initial_domain: %s"
+          ", provided target_domain: %s", initial_domain, target_domain);
+
+      ONNX_ASSERTM(strcmp(initial_version.domain().c_str(), target_version.domain()
+            .c_str()) == 0, "initial_version and target_version must have the same "
+          "domains");
+      for (auto it = mp_in.opset_import().begin(); it != mp_in.opset_import()
+          .end(); ++it) {
+        if (it->domain() == initial_version.domain()) {
+          ONNX_ASSERTM(initial_version.version() == it->version(),
+              "initial_version does not reflect current state of model");
+        }
+      }
+
+      std::shared_ptr<Graph> g(ImportModelProto(mp_in));
+      ONNX_ASSERTM(g.get() != nullptr,
+        "Warning: onnx version converter is unable to parse input model. "
+        "(The IR version of the ONNX model may be too old.)");
+
+      // TODO: Move to Inter-Domain Converter
+      // Get initial model versions
+      // std::vector<OpSetID> initial_versions = g->opset_versions();
+
+      // No conversion necessary if Model has single, equivalent opset version
+      // if (initial_versions.size() == 1 && initial_versions[0].version ==
+      //    target_version.version && initial_versions[0].domain ==
+      //    target_version.domain) {
+      //  return mp_in;
+      // }
+
+      // Check if target_version is valid
+      const std::unordered_map<std::string, std::pair<int, int>>& versions_map = OpSchemaRegistry::DomainToVersionRange::Instance().Map();
+      const std::string& search_domain = target_version.domain() == "ai.onnx" ? "" : target_version.domain();
```

Comment:
Here you're also storing a reference to a dead object.

Reply:
search_domain, right? (not versions_map)